crates/compiler/can/src/desugar.rs

```diff
@@ -607,57 +616,61 @@ pub fn desugar_expr<'a>(
+                        closure_arg_from_field(field_data[field_data.len() - 2]);
+                    let last_arg = closure_arg_from_field(field_data[field_data.len() - 1]);
 
                     let mut second_arg = Pattern::Tuple(Collection::with_items(
                         arena.alloc_slice_copy(&[second_to_last_arg, last_arg]),
                     ));
                     let mut second_arg_region =
                         Region::span_across(&second_to_last_arg.region, &last_arg.region);
 
-                    for index in (1..(field_names.len() - 2)).rev() {
+                    for index in (1..(field_data.len() - 2)).rev() {
                         second_arg =
                             Pattern::Tuple(Collection::with_items(arena.alloc_slice_copy(&[
-                                closure_arg_from_field(field_names[index]),
+                                closure_arg_from_field(field_data[index]),
                                 Loc::at(second_arg_region, second_arg),
                             ])));
                         second_arg_region =
-                            Region::span_across(&field_names[index].region, &second_arg_region);
+                            Region::span_across(&field_data[index].0.region, &second_arg_region);
                     }
 
                     arena.alloc_slice_copy(&[
-                        closure_arg_from_field(field_names[0]),
+                        closure_arg_from_field(field_data[0]),
                         Loc::at(second_arg_region, second_arg),
                     ])
                 }
             };
 
             let record_val = Record(Collection::with_items(
                 Vec::from_iter_in(
-                    field_names.iter().map(|field_name| {
-                        Loc::at(
-                            field_name.region,
-                            AssignedField::RequiredValue(
-                                Loc::at(field_name.region, field_name.value),
-                                &[],
-                                arena.alloc(Loc::at(
-                                    field_name.region,
-                                    Expr::Var {
-                                        module_name: "",
-                                        ident: arena.alloc_str(&format!("#{}", field_name.value)),
-                                    },
-                                )),
-                            ),
-                        )
-                    }),
+                    field_data
+                        .iter()
+                        .filter(|(_name, _val, ignored)| !ignored)
+                        .map(|(field_name, _val, _ignored)| {
+                            Loc::at(
+                                field_name.region,
+                                AssignedField::RequiredValue(
+                                    Loc::at(field_name.region, field_name.value),
```

Comment:
If both the `region` and `value` are the same, can't you just use `field_name` here?